packages/block-editor/src/components/block-tools/insertion-point.js

```diff
@@ -229,9 +230,25 @@ function InsertionPointPopover( {
 }
 
 export default function InsertionPoint( props ) {
-	const isVisible = useSelect( ( select ) => {
-		return select( blockEditorStore ).isBlockInsertionPointVisible();
+	const { insertionPoint, isVisible } = useSelect( ( select ) => {
+		const { getBlockInsertionPoint, isBlockInsertionPointVisible } =
+			select( blockEditorStore );
+		return {
+			insertionPoint: getBlockInsertionPoint(),
+			isVisible: isBlockInsertionPointVisible(),
+		};
 	}, [] );
 
-	return isVisible && <InsertionPointPopover { ...props } />;
+	return (
+		isVisible &&
+		( insertionPoint.operation === 'replace' ? (
+			<BlockDropZonePopover
+				// Force remount to trigger the animation.
+				key={ `${ insertionPoint.rootClientId }-${ insertionPoint.index }` }
+				{ ...props }
+			/>
+		) : (
+			<InbetweenInsertionPointPopover { ...props } />
```

Comment:
I think right now this is ignoring the "isVisible" check right? Should we return early if "! isVisible" instead?

Reply:
I think `isVisible` is at [L243](https://github.com/WordPress/gutenberg/pull/44647/files/43b266eaa21f322a688e1dc39fccb08c80aad5ee#diff-5a88c3390c631539a539a69de522717ef1e0fc69900524a458dd225b20c6b2aeR243)?

Reply:
You're right I mis-read the parenthesis.